iceoryx_dust/test/moduletests/test_cxx_string_conversion.cpp

```diff
@@ -0,0 +1,110 @@
+// Copyright (c) 2019 by Robert Bosch GmbH. All rights reserved.
```

Comment:
The test in this file were moved.

Reply:
@elfenpiff No worries at all. I understand the pain that you're feeling, it's cumbersome :upside_down_face: 

As IANAL, I'm going for option 2. and just checked who committed the tests. Turns out they were actually written by Bosch employees, so I'll keep the copyright.

```
244f5a3575 iceoryx_utils/test/moduletests/test_cxx_string.cpp (Hoinkis Simon (CC-AD/ESW1)  2020-03-21 20:29:47 +0100  334) TYPED_TEST(stringTyped_test, UnsafeSTDStringToStringConvConstrWithSize0ResultsInSize0)
244f5a3575 iceoryx_utils/test/moduletests/test_cxx_string.cpp (Hoinkis Simon (CC-AD/ESW1)  2020-03-21 20:29:47 +0100  335) {
8e1a8d3741 iceoryx_hoofs/test/moduletests/test_cxx_string.cpp (Simon Hoinkis               2022-01-12 17:10:41 +0100  336)     ::testing::Test::RecordProperty("TEST_ID", "83e1b7b2-8487-4c71-ac86-f4d5d98c1918");
23a6eaaa0f iceoryx_utils/test/moduletests/test_cxx_string.cpp (Lehmann Marika (CC-AD/ESW1) 2020-09-09 16:19:46 +0200  337)     using MyString = typename TestFixture::stringType;
2eabe1dd1c iceoryx_hoofs/test/moduletests/test_cxx_string.cpp (Simon Hoinkis               2021-12-02 17:07:28 +0100  338)     constexpr auto STRINGCAP = MyString::capacity();
244f5a3575 iceoryx_utils/test/moduletests/test_cxx_string.cpp (Hoinkis Simon (CC-AD/ESW1)  2020-03-21 20:29:47 +0100  339)     std::string testString;
79f1d66060 iceoryx_utils/test/moduletests/test_cxx_string.cpp (Marika Lehmann              2020-08-11 00:47:04 +0200  340)     string<STRINGCAP> fuu(TruncateToCapacity, testString);
79f1d66060 iceoryx_utils/test/moduletests/test_cxx_string.cpp (Marika Lehmann              2020-08-11 00:47:04 +0200  341)     EXPECT_THAT(fuu.capacity(), Eq(STRINGCAP));
3446dfe750 iceoryx_utils/test/moduletests/test_cxx_string.cpp (Lehmann Marika (CC-AD/ESW1) 2020-09-10 14:48:29 +0200  342)     EXPECT_THAT(fuu.size(), Eq(0U));
244f5a3575 iceoryx_utils/test/moduletests/test_cxx_string.cpp (Hoinkis Simon (CC-AD/ESW1)  2020-03-21 20:29:47 +0100  343)     EXPECT_THAT(fuu.c_str(), StrEq(""));
244f5a3575 iceoryx_utils/test/moduletests/test_cxx_string.cpp (Hoinkis Simon (CC-AD/ESW1)  2020-03-21 20:29:47 +0100  344) }

```